test/LibraryManager.IntegrationTest/VisualStudioLibmanHostTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Editor;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest
+{
+    [TestClass]
+    public class VisualStudioLibmanHostTest : VisualStudioHostTest
+    {
+        // Solution consts
+        const string _testSolutionName = @"TestSolution.sln";
+        const string _rootDirectoryName = @"TestSolution";
+
+        private static VisualStudioLibmanHostTest _instance;
+        private static string _resultPath;
+        private static string _solutionRootPath;
+        private static string _solutionPath;
+
+        protected override void DoHostTestInitialize()
+        {
+            _instance = this;
+
+            base.DoHostTestInitialize();
+
+            Solution.Open(_solutionPath);
```

Comment:
I assume these are no-ops if the solution is already open? Do you need to wait for restore at all?

Reply:
The DoHostTestCleanup function will close the solution.
Editors has some mechanism to keep solution open for different tests. I think we can consider adding that part later for solutions would growing larger.

Reply:
Did you do any verification against not re-opening the solution if it's already open?

---
In reply to: [192152520](https://github.com/aspnet/LibraryManager/pull/90#discussion_r192152520) [](ancestors = 192152520)

Reply:
This need some testService Implement elsewhere. I will open a bug for it to future optimize it.